src/execution/sql/deleter.cpp

```diff
@@ -0,0 +1,50 @@
+#include "execution/sql/deleter.h"
+#include <algorithm>
+#include "execution/exec/execution_context.h"
+#include "execution/util/execution_common.h"
+
+namespace terrier::execution::sql {
+
+Deleter::Deleter(exec::ExecutionContext *exec_ctx, catalog::table_oid_t table_oid)
+    : table_oid_(table_oid), exec_ctx_(exec_ctx) {
+  table_ = exec_ctx->GetAccessor()->GetTable(table_oid);
+
+  max_pr_size_ = 0;
+  auto index_oids = exec_ctx->GetAccessor()->GetIndexOids(table_oid_);
+  for (auto index_oid : index_oids) {
+    max_pr_size_ = std::max(max_pr_size_, GetIndex(index_oid)->GetProjectedRowInitializer().ProjectedRowSize());
+  }
+  index_pr_buffer_ = exec_ctx->GetMemoryPool()->AllocateAligned(max_pr_size_, sizeof(uint64_t), true);
+}
+
+Deleter::~Deleter() { exec_ctx_->GetMemoryPool()->Deallocate(index_pr_buffer_, max_pr_size_); }
+
+common::ManagedPointer<storage::index::Index> execution::sql::Deleter::GetIndex(catalog::index_oid_t index_oid) {
+  auto iter = index_cache_.find(index_oid);
```

Comment:
I don't think we should be caching catalog lookups at this level. We should cache at the Catalog level.